objectivec/GPBMessage_PackagePrivate.h

```diff
@@ -42,6 +44,125 @@
 
 #import "GPBBootstrap.h"
 
+#if defined(__LP64__) && __LP64__
+#define GPB_ASM_PTR " .quad "
+#define GPB_ASM_PTRSIZE " 8 "
+#define GPB_ASM_ONLY_LP64(x) x
+#define GPB_ALIGN_SIZE " 3 "
+#define GPB_DESCRIPTOR_METHOD_TYPE "@16@0:8"
+#else  // __LP64__
+#define GPB_ASM_PTR " .long "
+#define GPB_ASM_PTRSIZE " 4 "
+#define GPB_ASM_ONLY_LP64(x)
+#define GPB_ALIGN_SIZE " 2 "
+#define GPB_DESCRIPTOR_METHOD_TYPE "@8@0:4"
+#endif  // __LP64__
+
+#define GPB_MESSAGE_CLASS_NAME_RAW GPBMessage
+#define GPB_MESSAGE_CLASS_NAME GPBStringifySymbol(GPB_MESSAGE_CLASS_NAME_RAW)
+
+#if !__has_feature(objc_arc)
+#define GPB_CLASS_FLAGS " 0x00 "
+#define GPB_METACLASS_FLAGS " 0x01 "
+#else
+// 0x80 denotes that the class supports ARC.
+#define GPB_CLASS_FLAGS " 0x80 "
+#define GPB_METACLASS_FLAGS " 0x81 "
+#endif
+
+// This generates code equivalent to:
+// ```
+// @implementation _name_
+// + (GPBMessage*)descriptor {
+//      return _descriptorFunc_(self, _cmd);
+// }
+// @end
+// ```
+// We do this to avoid all of the @property metadata.
+// Note that for Xcode < 12 having the cstring_literals be local definitions
+// is required to avoid a "malformed method list" assertion from the linker.
+// We do this by starting them with "L":
+// https://ftp.gnu.org/old-gnu/Manuals/gas-2.9.1/html_chapter/as_5.html#SEC48
+#define GPB_MESSAGE_SUBCLASS_IMPL(name, descriptorFunc)                        \
```

Comment:
Is it worth listed all the places explicitly to allow for cleanup eventually?

Reply:
Not sure what cleanup you mean? Xcode 12.5 declares them local as well (i.e. this is correct). The linker with 11.7 is just pickier about it. I'll update the comment.

Reply:
There.. see if that is clearer.

Reply:
Does 13 declare them as local also?

The current comment is written such that it sorta implies they only have to be local because of the assertion, not from a correctness pov.  If correctness is local, then I wouldn't both with the version comment at all.